File: 2020/day1.swift
import Foundation
do {
  let content = try String(contentsOfFile: "input.txt")
  let intArray = content.components(separatedBy: "\n").compactMap { Int($0) }

  let dict = intArray.reduce(into: [Int: Int]()) {
    $0[$1] = 2020 - $1
  }

  for entry in dict {
    for number in intArray {
      if intArray.contains(entry.value - number) {
        print(entry.key)
        print(number)
        print(entry.value - number)
        exit(-1)
      }
    }
  }
}

File: 2020/day1.dart
import 'dart:io';

main() {
  var path = 'input.txt';
  final file = new File(path);
  final List<int> list =
      file.readAsLinesSync().map((e) => int.parse(e)).toList();
  final map = new Map.fromIterable(list, key: (k) => k, value: (v) => 2020 - v);

  map.forEach((key, value) {
    for (var number in list) {
      if (list.contains(value - number)) {
        print('$key $number ${value - number}');
        print('${key * number * (value - number)}');
        exit(-1);
      }
    }
  });
}
